jobbergate-api/jobbergate_api/apps/job_submissions/schemas.py

```diff
@@ -268,12 +286,12 @@ class JobProperties(BaseModel, extra=Extra.forbid):
     threads_per_core: Optional[int] = Field(
         description="Restrict node selection to nodes with at least the specified number of threads per core."
     )
-    time_limit: Optional[str] = Field(description="Step time limit.")
+    time_limit: Optional[LengthLimitedStr] = Field(description="Step time limit.")
     time_minimum: Optional[int] = Field(description="Minimum run time in minutes.")
     wait_all_nodes: Optional[int] = Field(
```

Comment:
```suggestion
    wait_all_nodes: Optional[NonNegativeInt] = Field(
```

Reply:
I had to revert this one because it was already constrained `Field(ge=0, le=1)`, so `NonNegativeInt` results in an error.